src/utilities.ts

```diff
@@ -43,6 +44,50 @@ export function getMinorFullName(acronym: string): string {
   return requirementJSON.minor[acronym].name;
 }
 
+export function getAllSubjects(): ReadonlySet<string> {
+  const set = new Set<string>();
+  fullCoursesArray.forEach(it => set.add(it.subject));
+  return set;
+}
+
+const SUBJECT_COLORS = [
+  {
+    text: 'Red',
+    hex: 'DA4A4A',
+  },
+  {
+    text: 'Orange',
+    hex: 'FFA53C',
+  },
+  {
+    text: 'Green',
+    hex: '58C913',
+  },
+  {
+    text: 'Blue',
+    hex: '139DC9',
+  },
+  {
+    text: 'Purple',
+    hex: 'C478FF',
+  },
+  {
+    text: 'Pink',
+    hex: 'F296D3',
+  },
+];
+
+export function allocateSubjectColor(subjectColors: Record<string, string>, subject: string): void {
+  if (subjectColors[subject]) return;
+  const usedColors = new Set(Object.values(subjectColors));
+  // Filter out used colors
+  const unusedSubjectColors = SUBJECT_COLORS.filter(color => !usedColors.has(color.hex));
+  const subjectColorCandidates =
+    unusedSubjectColors.length !== 0 ? unusedSubjectColors : SUBJECT_COLORS;
+  subjectColors[subject] =
+    subjectColorCandidates[Math.floor(Math.random() * subjectColorCandidates.length)].hex;
+}
```

Comment:
Moved from `global-firestore-data` and rewritten. Added a test to ensure correctness.